sm_property_tree/include/sm/BoostPropertyTree.hpp

```diff
@@ -45,9 +45,11 @@ namespace sm {
 
     /**
      * Update this ptree with another, allowing merging of ptrees
-     * @param with
+     * @param with the other ptree
+     * @param createIfNecessary it true (default) all necessary trees will be created automatically to store the values of with.
+     * If not it will throw an KeyNotFoundException if there is no such key already
      */
-    void update(const BoostPropertyTree & with);
+    void update(const BoostPropertyTree & with, bool createIfNecessary = true);
```

Comment:
default values are discouraged. Can you rather make an overload and call through with default = true.